odl/util/testutils.py

```diff
@@ -251,19 +251,37 @@ def _pass(function):
 # Helpers to generate data
 def example_array(space):
     """Generate an example array that is compatible with ``space``."""
-    # Generate numpy vectors, real or complex or int
-    if np.issubdtype(space.dtype, np.floating):
-        arr = np.random.randn(space.size)
-    elif np.issubdtype(space.dtype, np.integer):
-        arr = np.random.randint(-10, 10, space.size)
+    from odl.space import ProductSpace
+    if isinstance(space, ProductSpace):
+        return np.array([example_array(si) for si in space])
     else:
-        arr = np.random.randn(space.size) + 1j * np.random.randn(space.size)
+        # Generate numpy vectors, real or complex or int
+        if np.issubdtype(space.dtype, np.floating):
+            arr = np.random.randn(space.size)
+        elif np.issubdtype(space.dtype, np.integer):
+            arr = np.random.randint(-10, 10, space.size)
+        else:
+            arr = (np.random.randn(space.size) +
+                   1j * np.random.randn(space.size))
 
-    return arr.astype(space.dtype, copy=False)
+        return arr.astype(space.dtype, copy=False)
 
 
 def example_element(space):
-    return space.element(example_array(space))
+    """Construct an example element in a space.
+
+    The element is chosen to be a totally random vector.
```

Comment:
While at it, why not call it `random_element`? "example" suggests something reasonable.